mission/internal_status/bootscripts/OLED.py

```diff
@@ -135,37 +116,26 @@ def func_check(func):
 
 
 def read_PSM_voltage():
-    # Read ADC values a set amount and take average to make signal more stable
-    voltage = 0
-    for i in range(average_of):
-        # Sometimes an I/O timeout or error happens, it will run again when the error disappears
-        try:
-            voltage += (
-                channel_voltage.get_voltage_from_reading() * psm_to_battery_voltage
-            )
+    # Sometimes an I/O timeout or error happens, it will run again when the error disappears
+    try:
+        voltage = channel_voltage.convert_and_read() * psm_to_battery_voltage
 
-        except IOError:
-            voltage = 0
-            break
+    except IOError:
+        voltage = 0
```

Comment:
Returning zero on error means that there is no way to separate disconnect (i.e. actually no voltage) and I2C errors. Might be better to return -1 or something along those lines :cat:

Reply:
Done :)